packages/plugin-kit/README.md

```diff
@@ -205,6 +205,8 @@ The `TextSourceCodeBase` class is intended to be a base class that has several o
 
 - `lines` - an array of text lines that is created automatically when the constructor is called.
 - `getLoc(node)` - gets the location of a node. Works for nodes that have the ESLint-style `loc` property and nodes that have the Unist-style [`position` property](https://github.com/syntax-tree/unist?tab=readme-ov-file#position). If you're using an AST with a different location format, you'll still need to implement this method yourself.
+- `getLocFromIndex(index)` - Converts a source text index into a `{ line: number, column: number }` pair.
+- `getIndexFromLoc(loc)` - Converts a `{ line: number, column: number }` pair into a source text index.
```

Comment:
Perhaps we should add a note that, in order for these to work, the root node must always cover the entire source code text?